src/aws-cpp-sdk-core/include/aws/core/client/ClientConfiguration.h

```diff
@@ -465,6 +465,16 @@ namespace Aws
                */
               bool useAnonymousAuth = false;
             } winHTTPOptions;
+
+          /**
+            * The AWS AccountId used for the request.
+            */
+          Aws::String accountId;
+
+          /**
+           * The AccountId Endpoint Mode.
```

Comment:
the comparison is done by
```
strcmp(mapping.first, accountIdMode.c_str()) == 0; });
```

i.e. not case insensitive.

In an ideal world, I'd suggest to use an enum here.
On practice, please just add a comment with possible allowed values and the requirements of the lowercase.

Reply:
I agree this is a better user experience, however this would be backwards breaking because on main because of how codegen is currently generating it.

```cpp
#include <aws/core/Aws.h>
#include <aws/dynamodb/DynamoDBClient.h>

using namespace Aws;
using namespace Aws::DynamoDB;

auto main() -> int {
  SDKOptions options;
  InitAPI(options);
  {
    DynamoDBClientConfiguration config;
    config.accountIdEndpointMode = "preferred";
    DynamoDBClient client{config};
  }
  ShutdownAPI(options);
  return 0;
}
```

> In an ideal world/On practice

yep yep, will add comment regarding